hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactUtil.java

```diff
@@ -165,4 +173,44 @@ public static Short[] characterArrayAsShortArray(Character[] characterArray) {
         }
         return shortArray;
     }
+
+    public static void verifyClassIsCompactSerializable(Class<?> clazz) {
+        if (canBeSerializedAsCompact(clazz)) {
+            return;
+        }
+
+        throw new HazelcastSerializationException("The '" + clazz + "' cannot "
+                + "be serialized with zero configuration Compact serialization "
+                + "because this type is not supported yet. If you want to "
+                + "serialize this class, consider writing a CompactSerializer "
+                + "for it.");
+    }
+
+    public static void verifyFieldClassIsCompactSerializable(Class<?> fieldClass, Class<?> clazz) {
+        if (canBeSerializedAsCompact(fieldClass)) {
+            return;
+        }
+
+        throw new HazelcastSerializationException("The '" + fieldClass + "' "
+                + "cannot be serialized with zero configuration Compact "
+                + "serialization because this type is not supported yet. If you "
+                + "want to serialize '" + clazz + "' which uses this class in "
+                + "its fields, consider writing a CompactSerializer for it.");
+    }
+
+    private static boolean canBeSerializedAsCompact(Class<?> clazz) {
+        Package classPackage = clazz.getPackage();
+        if (classPackage == null) {
```

Comment:
If class is int global namespace, what is this value? We probably need to support this case.

Reply:
yes, that is what I meant.